src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -280,29 +288,33 @@ public final void apply(Project project) {
             });
             fullLockStateProperty.set(project.provider(fullLockStateSupplier::get));
 
-            if (project.getGradle().getStartParameter().isWriteDependencyLocks()) {
-                if (isSkipWriteLocks(project)) {
-                    log.lifecycle(
-                            "Skipped writing lock state to {} because the 'gcvSkipWriteLocks' property was set",
-                            rootLockfile);
+            project.getGradle().getTaskGraph().whenReady(taskExecutionGraph -> {
```

Comment:
Ok, I've made a little change here which _I think_ should be safe (but as always with Gradle...). We wait for the task graph to be evaluated before writing the lock file, so that abbreviating task names still works (like `wVL`, `wVersionLock`) which should make dev's lives a little easier.

Reply:
Unsurprisingly this does not work. I think task graph evaluation time is too late.